src/nvenc/nvenc_base.cpp

```diff
@@ -321,6 +321,12 @@ namespace nvenc {
         set_ref_frames(format_config.maxNumRefFramesInDPB, format_config.numRefL0, 5);
         set_minqp_if_enabled(config.min_qp_hevc);
         fill_h264_hevc_vui(format_config.hevcVUIParameters);
+        if (client_config.enableIntraRefresh == 1) {
+          format_config.enableIntraRefresh = 1;
+          format_config.singleSliceIntraRefresh = 1;

```

Comment:
I think `singleSliceIntraRefresh` is a special feature that's not supported by all NVENC encoders. Do we need that?

Reply:
It's possible to check for `NV_ENC_CAPS_SINGLE_SLICE_INTRA_REFRESH` cap and enable single slice conditionally.

Reply:
Done!